src/main/java/org/zalando/nakadi/controller/SubscriptionStreamController.java

```diff
@@ -110,10 +110,8 @@ public void onException(final Exception ex) {
         }
 
         @Override
-        public void streamData(final byte[] data) throws IOException {
-            headersSent = true;
```

Comment:
@antban it wasn't clear why `headersSent` should be set in this method. It's already set in line 96 in the method above `onException` and also in line 84 in method `onInitialized`. It was not clear which scenario required it to be set in the `streamData`.

Also, this new streaming code requires access to the output stream directly, since it writes arrays of bytes to it directly. This code is shared between low level and high level, and only high level encapsulates the output stream.